after upgrading to latest version server cannot get started throwing exception

## Information

- **Version:** 6.114.8
- Packages:
getting this error attaching stacktrace
```
[2022-05-12T04:06:16.578] [ERROR] [TSED] - TypeError: Cannot read properties of undefined (reading 'isStore')
    at Function.from (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/domain/JsonEntityStore.ts:215:17)
    at getJsonEntityStore (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/utils/getJsonEntityStore.ts:34:26)
    at JsonLazyRef.get schema [as schema] (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/domain/JsonLazyRef.ts:16:30)
    at lazyRefMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/lazyRefMapper.ts:18:22)
    at execMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/registries/JsonSchemaMapperContainer.ts:35:35)
    at anyMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/anyMapper.ts:13:22)
    at execMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/registries/JsonSchemaMapperContainer.ts:35:35)
    at itemMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/itemMapper.ts:5:83)
    at execMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/registries/JsonSchemaMapperContainer.ts:35:35)
    at /home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/ofMapper.ts:7:22 {
  name: 'INJECTION_ERROR',
  tokens: [ Symbol(UsersSchema) ],
  origin: TypeError: Cannot read properties of undefined (reading 'isStore')
      at Function.from (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/domain/JsonEntityStore.ts:215:17)
      at getJsonEntityStore (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/utils/getJsonEntityStore.ts:34:26)
      at JsonLazyRef.get schema [as schema] (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/domain/JsonLazyRef.ts:16:30)
      at lazyRefMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/lazyRefMapper.ts:18:22)
      at execMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/registries/JsonSchemaMapperContainer.ts:35:35)
      at anyMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/anyMapper.ts:13:22)
      at execMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/registries/JsonSchemaMapperContainer.ts:35:35)
      at itemMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/itemMapper.ts:5:83)
      at execMapper (/home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/registries/JsonSchemaMapperContainer.ts:35:35)
      at /home/shivansh/WebstormProjects/BlendtrixBackend/node_modules/@tsed/schema/src/components/ofMapper.ts:7:22
}
```

--- a/packages/orm/mongoose/src/decorators/ref.spec.ts
+++ b/packages/orm/mongoose/src/decorators/ref.spec.ts
@@ -1,11 +1,25 @@
-import {Store} from "@tsed/core";
+import {catchError, Store} from "@tsed/core";
 import {getJsonSchema, Property} from "@tsed/schema";
 import {Schema} from "mongoose";
 import {MONGOOSE_MODEL_NAME, MONGOOSE_SCHEMA} from "../constants/constants";
-import {Ref} from "./ref";
 import {MongooseModels} from "../registries/MongooseModels";
+import {Ref} from "./ref";
 
 describe("@Ref()", () => {
+  describe("type is a class", () => {
+    it("should set metadata", () => {
+      const error = catchError(() => {
+        class Model {
+          @Ref(undefined)
+          num: number[];
+        }
+      });
+
+      expect(error?.message).toEqual(
+        "A model is required on `@Ref(model)` decorator. Please give a model or wrap it inside an arrow function if you have a circular reference."
+      );
+    });
+  });
   describe("type is a class", () => {
     it("should set metadata", () => {
       class RefTest {

--- a/packages/orm/mongoose/src/decorators/ref.ts
+++ b/packages/orm/mongoose/src/decorators/ref.ts
@@ -63,6 +63,12 @@ export function Ref(
   model: string | (() => Type) | any,
   options: RefOptions | MongooseSchemaTypes = MongooseSchemaTypes.OBJECT_ID
 ): PropertyDecorator {
+  if (!model) {
+    throw new Error(
+      "A model is required on `@Ref(model)` decorator. Please give a model or wrap it inside an arrow function if you have a circular reference."
+    );
+  }
+
   const getType = () => (isString(model) ? MongooseModels.get(model) : isArrowFn(model) ? model() : model);
   const populatedGroups = (isObject(options) && options.populatedGroups) || [];
 

--- a/packages/specs/schema/src/decorators/collections/collectionOf.spec.ts
+++ b/packages/specs/schema/src/decorators/collections/collectionOf.spec.ts
@@ -1,9 +1,23 @@
+import {catchError} from "@tsed/core";
 import {getJsonSchema, In, MaxItems, MinItems, OperationPath, Property} from "@tsed/schema";
 import {JsonEntityStore} from "../../domain/JsonEntityStore";
 import {CollectionContains} from "./collectionContains";
 import {ArrayOf, CollectionOf, MapOf} from "./collectionOf";
 
 describe("@CollectionOf", () => {
+  it("should declare a collection (Array of)", () => {
+    // WHEN
+    const error = catchError(() => {
+      class Model {
+        @CollectionOf(undefined)
+        num: number[];
+      }
+    });
+
+    expect(error?.message).toEqual(
+      "A type is required on `@CollectionOf(type)` decorator. Please give a type or wrap it inside an arrow function if you have a circular reference."
+    );
+  });
   it("should declare a collection (Array of)", () => {
     // WHEN
     class Model {

--- a/packages/specs/schema/src/decorators/collections/collectionOf.ts
+++ b/packages/specs/schema/src/decorators/collections/collectionOf.ts
@@ -98,12 +98,17 @@ export interface CollectionOfChainedDecorators extends MapOfChainedDecorators, A
  * @collections
  */
 export function CollectionOf(type: any, collectionType?: any): CollectionOfChainedDecorators {
+  if (!type) {
+    throw new Error(
+      "A type is required on `@CollectionOf(type)` decorator. Please give a type or wrap it inside an arrow function if you have a circular reference."
+    );
+  }
+
   const schema: any = {};
   let contains: boolean = false;
 
   const decorator = (...args: any) => {
     const store = JsonEntityStore.from(...args);
-    // const itemSchema = store.itemSchema.toJSON();
 
     if (collectionType) {
       store.collectionType = collectionType;
@@ -111,10 +116,8 @@ export function CollectionOf(type: any, collectionType?: any): CollectionOfChain
     }
 
     store.type = type;
-    // console.log(type);
     store.itemSchema.type(type);
-    // console.log(store.itemSchema.getComputedType(), schema);
-    // store.itemSchema.assign({...itemSchema, type});
+
     store.schema.assign(schema);
 
     if (store.isArray && contains) {

